### PR TITLE
HVN route migration guide

### PR DESCRIPTION
**Pull request by bcmdarroch**
### :hammer_and_wrench: Description

Adds a migration guide that captures the steps in https://github.com/hashicorp/terraform-provider-hcp/pull/128 for adding HVN routes to existing HCP networking setup managed by Terraform.

### CODE REVIEW

File: docs/guides/hvn-route-migration-guide.md [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+page_title: "HVN Route Migration Guide - HCP Provider"
+description: |-
+    An guide to migrating HCP networking resources to use HVN routes.
+---
+
+# Introducing HVN routes
+
+The HVN route is a new resource that belongs to an HVN. It contains a CIDR block and targets a networking connection: 
+currently, either a peering or transit gateway attachment.
+TODO: expand on why they're being introduced.
```
Comment (bcmdarroch) [inline docs/guides/hvn-route-migration-guide.md:12]: I could use some help on what to say here. Probably a sentence or two on why this change is being introduced and its benefits.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,10 @@
+## Unreleased ()
+
+FEATURES:
+* **New resource** `hcp_hvn_route` (#122)
+
+⚠️ Note: Due to the introduction of HVN routes, this version introduces breaking changes to the `hcp_aws_network_peering` and `hcp_aws_transit_gateway_attachment` resources. Please see our [migration guide](link TODO) to learn about how to update your HCP networking setup to work with HVN routes. ⚠️
```
Comment (smaant) [inline CHANGELOG.md:6]: do we a link already?

File: docs/guides/hvn-route-migration-guide.md (hunk 1 above)
Comment (smaant) [reply docs/guides/hvn-route-migration-guide.md:12]: I would say something like: "HVN routes provide a general view on how HVN traffic is routed across all networking connections and create a flexible way of managing these routing rules."

**Review by smaant — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (bcmdarroch) [reply CHANGELOG.md:6]: Yeah I'm going to link to your PR with the migration steps here, since I don't think we'll keep the migration guide in the registry docs forever.
